test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -362,7 +362,60 @@ static public void testWithFieldOnNonExistentClass() throws Throwable {
 			Assert.fail("should throw error. Class does not exist");
 		} catch (NoClassDefFoundError e) {}
 	}
-	
+
+	/*
+	 * Test ifacmp on value class
+	 * 
+	 * class TestIfacmpOnValueClass {
+	 *  
+	 * }
+	 */
+	@Test(priority=1)
+	static public void TestIfacmpOnValueClass() throws Throwable {
+		String fields[] = {"x:I", "y:I"};
+		point2DClass = ValueTypeGenerator.generateValueClass("Point2DInAcmp", fields);
+		makePoint2D = lookup.findStatic(point2DClass, "makeValue", MethodType.methodType(point2DClass, int.class, int.class));
+
+		int x = 0;
+		int y = 0;
+		
+		Object valueType = makePoint2D.invoke(x, y);
+
+		Object refType = (Object) x;
+		
+		if (valueType == refType) {
+			Assert.fail("should throw error. Acmpne passes one valueType and one refType!");
```

Comment:
`should throw error` is misleading.

It doesnt throw an error. Its probably simpler if you dont mention acmp in the test. You can simply say:

"An identity (==) comparison that contains a valueType should always return false" and vice versa for the "!=" case